### ISSUE TITLE
resource_block to lazy a notifying_block

### ISSUE THREAD

**Issue by lamont-granquist** [chef/chef#4978]
notifying_block is a DSL method so at compile-time it creates a subresource collection and eagerly converges it.

to lazy the execution of the subcollection we could offer a primitive like this:

``` ruby
resource_block "stuff" do
  block do
    template "/etc/whatever/thing.conf1" do
      source "thing.conf1.erb"
      notifies :restart, "service[whatever]"
    end
    template "/etc/whatever/thing.conf2" do
      source "thing.conf2.erb"
      notifies :restart, "service[whatever]"
    end
    service "whatever" do
      action :start
    end
  end
  action :run
end
```

of course i believe this is entirely equivalent to:

``` ruby
whyrun_safe_ruby_block "stuff" do
  block do
    nofiying_block do
      template "/etc/whatever/thing.conf1" do
        source "thing.conf1.erb"
        notifies :restart, "service[whatever]"
      end
      template "/etc/whatever/thing.conf2" do
        source "thing.conf2.erb"
        notifies :restart, "service[whatever]"
      end
      service "whatever" do
        action :start
      end
    end
  end
  action :run
end
```

But I can potentially see a lot of repetitive conversations about the 'right' way to do the latter and it might be simpler to just be able to say `use a resource_block, it does exactly what you need...`

**Comment by lamont-granquist** [comment]
@coderanger thoughts?  up? down?  meh?

**Comment by coderanger** [comment]
I don't love the syntax :-/

I think something like this is probably more grokkable:

```
recipe 'whatever' do
 template '...' do end
  file '...' etc etc
end
```

This is further from the usual resource syntax but probably more easily understandable in the short term.

**Comment by coderanger** [comment]
I'm also not totally sold that this is enough better than the existing LWRP stuffs to merit syntax.

**Comment by lamont-granquist** [comment]
Yeah, I was thinking about going down that road.    The thing in the block doesn't quite act like a resource when it comes to the content of the block though.  Instead of instance_eval'ing the block against the resource, we probably need to have a special option to save the whole block for use by the provider later.   Not that hard to code, but it starts to abuse the 'shape' of what a resource is.

**Comment by lamont-granquist** [comment]
closing this since we closed #4982
